src/Extend/Model.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Extend;
+
+use Flarum\Extension\Extension;
+use Illuminate\Contracts\Container\Container;
+
+class Model implements ExtenderInterface
+{
+    private $modelClass;
+    private $dateCallbacks = [];
+    private $defaultAttributeCallbacks = [];
+    private $relationships = [];
+
+    /**
+     * @param string $modelClass The ::class attribute of the model you are modifying.
+     *                           This model should extend from \Flarum\Database\AbstractModel.
+     */
+    public function __construct(string $modelClass)
+    {
+        $this->modelClass = $modelClass;
+    }
+
+    /**
+     * Modify which attributes of this model are treated as dates.
+     *
+     * @param callable $callable
+     *
+     * The callable can be a closure or invokable class, and should accept:
+     * - \Flarum\User\User $user: the user in question.
+     * - array $dateAttributes: an array of attributes which are processed as dates.
+     *
+     * The callable should return:
+     * - array $dateAttributes: an array of attributes which are processed as dates.
+     */
+    public function configureDates(callable $callable)
+    {
+        $this->dateCallbacks[] = $callable;
+
+        return $this;
+    }
+
+    /**
+     * Modify default attribute values for this model.
+     *
+     * @param callable $callable
+     *
+     * The callable can be a closure or invokable class, and should accept:
+     * - \Flarum\User\User $user: the user in question.
+     * - array[string] $dateAttributes: an associative array of attributes => default values.
+     *
+     * The callable should return:
+     * - array[string] $dateAttributes: an associative array of attributes => default values.
+     */
+    public function configureDefaultAttributes(callable $callable)
```

Comment:
Same here, I would prefer `defaultAttribute($name, $value)` here. `$value` could either be a static value (for simple scalars) or a callable. 

Naming: What do you think about `defaultValue(...)` instead?

Reply:
I put it to just `->default($attribute, $value)` for now